plugins/builds/steps/update.js

```diff
@@ -20,43 +20,66 @@ module.exports = () => ({
             }
         },
         handler: (request, reply) => {
-            const factory = request.server.app.buildFactory;
+            const buildFactory = request.server.app.buildFactory;
+            const stepFactory = request.server.app.stepFactory;
             const buildId = request.params.id;
+            const stepName = request.params.name;
             const buildIdCred = request.auth.credentials.username;
             let stepIndex = -1;
 
             if (buildId !== buildIdCred) {
                 return reply(boom.forbidden(`Credential only valid for ${buildIdCred}`));
             }
 
-            return factory.get(buildId)
+            // Make sure build exists
+            return buildFactory.get(buildId)
                 .then((build) => {
                     if (!build) {
                         throw boom.notFound('Build does not exist');
                     }
-                    const steps = build.steps;
                     const now = (new Date()).toISOString();
 
-                    stepIndex = steps.findIndex(step => step.name === request.params.name);
+                    // Check if step model exists
+                    return stepFactory.get({ buildId, name: stepName })
+                        .then((step) => {
+                            if (!step) {
```

Comment:
will this happen, once new code goes in ? Is this specifically for builds which were created before this release and still running after release ?

Reply:
Yes, this is for backwards compatibility. I think it should only apply if a build is in the middle of running when this change is pulled in.